instrumentation/activity/src/main/java/io/opentelemetry/android/instrumentation/activity/ActivityLifecycleInstrumentation.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation.activity
+
+import android.app.Application
+import android.os.Build
+import com.google.auto.service.AutoService
+import io.opentelemetry.android.OpenTelemetryRum
+import io.opentelemetry.android.instrumentation.AndroidInstrumentation
+import io.opentelemetry.android.instrumentation.activity.startup.AppStartupTimer
+import io.opentelemetry.android.instrumentation.common.Constants.INSTRUMENTATION_SCOPE
+import io.opentelemetry.android.instrumentation.common.ScreenNameExtractor
+import io.opentelemetry.android.internal.services.ServiceManager
+import io.opentelemetry.android.internal.services.visiblescreen.VisibleScreenService
+import io.opentelemetry.android.internal.services.visiblescreen.activities.DefaultingActivityLifecycleCallbacks
+import io.opentelemetry.api.trace.Tracer
+
+@AutoService(AndroidInstrumentation::class)
+class ActivityLifecycleInstrumentation : AndroidInstrumentation {
+    private val startupTimer: AppStartupTimer by lazy { AppStartupTimer() }
+    private var screenNameExtractor: ScreenNameExtractor = ScreenNameExtractor.DEFAULT
+    private var tracerCustomizer: (Tracer) -> Tracer = { it }
+
+    fun setTracerCustomizer(customizer: (Tracer) -> Tracer) {
+        tracerCustomizer = customizer
+    }
+
+    fun setScreenNameExtractor(screenNameExtractor: ScreenNameExtractor) {
+        this.screenNameExtractor = screenNameExtractor
+    }
```

Comment:
I think it's totally fine for the purposes of a prototype, but I think we should be favoring immutability, and these should be injected rather than `.set()`.

Reply:
...and I'm not just being dogmatic. I think the setters can give the user a sense that they can change the instrumentation at runtime (although they probably won't have access to it anyway).